{
    "version": "1.0.0",
    "title": "System is resilient to provider's failures",
    "description": "Can our consumer survive gracefully a provider's failure?",
    "tags": [
        "service",
        "kubernetes",
        "spring"
    ],
     "configuration": {
        "app_name": {
		"type" : "env",
		"key" : "LABEL_NAME"
	},
        "name_space": {
        	"type": "env",
        	"key": "NAME_SPACE"
          },
        "service_url": {
            "type": "env",
            "key": "APP_ENDPOINT"
        }
    },
    "steady-state-hypothesis": {
        "title": "Application is normal",
        "probes": [
            {
                "type": "probe",
                "name": "application-must-respond-normally",
                "tolerance": 200,
                "provider": {
                    "type": "http",
                    "url": "https://${service_url}/health/full",
                    "timeout": 0.5
                }
            }
        ]
    },
    "method": [
	    {
            "type": "action",
            "name": "Terminate_pod",
            "provider": {
                "type": "python",
                "module": "chaosk8s.pod.actions",
                "func": "terminate_pods",
                "arguments": {
                    "label_selector": "app=${app_name}",
                    "name_pattern": "${app_name}",
                    "ns": "${name_space}",
                    "rand": true,
                    "mode": "fixed",
                    "qty": 1
                }
            },
            "pauses": {
                "after": 20
            }
        }
    ],
    "rollbacks": []
}
